source/MRMesh/MRMeshBoolean.cpp

```diff
@@ -27,9 +27,17 @@ BooleanResult boolean( const Mesh& meshA, const Mesh& meshB, BooleanOperation op
     bool needCutMeshB = opearation != BooleanOperation::InsideA && opearation != BooleanOperation::OutsideA;
 
     if ( needCutMeshA )
+    {
+        // build aabbtree for init mesh
```

Comment:
// build tree for input mesh for the cloned mesh to copy the tree,
// this is important for many calls to Boolean for the same mesh to avoid tree construction on every call